ggshield/core/config/config.py

```diff
@@ -123,7 +165,16 @@ def dashboard_url(self) -> str:
 
     @property
     def api_key(self) -> str:
+        return self.get_api_key()[0]
+
+    @property
+    def api_key_source(self) -> ConfigSource:
+        return self.get_api_key()[1]
+
+    def get_api_key(self) -> Tuple[str, ConfigSource]:
```

Comment:
I think you should do the same here as for the instance name: remove the property and rename `get_api_key()` to `get_api_key_and_source()`.